docs/architecture/adr-019-protobuf-state-encoding.md

```diff
@@ -221,23 +222,20 @@ every module that implements it in order to populate the `InterfaceRegistry`.
 
 ### Using `Any` to encode state
 
-The SDK will provide support methods `MarshalAny` and `UnmarshalAny` to allow
-easy encoding of state to `Any` in `Codec` implementations. Ex:
+The SDK will provide support methods `MarshalInterface` and `UnmarshalInterface` to hide a complexity of wrapping interface types into `Any` and allow easy serialization.
 
 ```go
 import "github.com/cosmos/cosmos-sdk/codec"
 
-func (c *Codec) MarshalEvidence(evidenceI eviexported.Evidence) ([]byte, error) {
-	return codec.MarshalAny(evidenceI)
+// note: eviexported.Evidence is an interface type
+func MarshalEvidence(cdc codec.BinaryMarshaler, e eviexported.Evidence) ([]byte, error) {
+	return cdc.MarshalInterface(e)
 }
 
-func (c *Codec) UnmarshalEvidence(bz []byte) (eviexported.Evidence, error) {
+func UnmarshalEvidence(cdc codec.BinaryMarshaler, bz []byte) (eviexported.Evidence, error) {
 	var evi eviexported.Evidence
-	err := codec.UnmarshalAny(c.interfaceContext, &evi, bz)
-	if err != nil {
-		return nil, err
-	}
-	return evi, nil
+	err := cdc.UnmarshalInterface(bz, &evi)
+    return err, nil
```

Comment:
shouldn't it be `evi, err`?

Reply:
Also, the indentation is weird

Reply:
Yes, I changed the order to a canonical one in a later commit. Thanks for catching.